decode.go

```diff
@@ -16,11 +16,15 @@ type DecodeFeedback interface {
 	// is shorter than internal layer variables (HeaderLength, or the like) say it
 	// should be.  It sets packet.Metadata().Truncated.
 	SetTruncated()
+	// SetConsumed should be called for streams to mention that only a part of
+	// the payload has been consumed (currently useful for TCP only).
+	SetConsumed(uint)
```

Comment:
I think the best way forward would be for us to provide a sample TCP layer which decodes in "packet mode" and uses this to notify if there's not enough data. Let us prepare something and get back to you.